absorption.py

```diff
@@ -0,0 +1,88 @@
+import scipp as sc
+import mantid.simpleapi as simpleapi
+
+def ensure_sample_in_ws(ws):
+    """
+    test for shape in sample - if not there, add a default vanadium cylinder
+
+    Parameters
+    ----------
+        ws: Mantid Workspace
+            Workspace with Sample defined
+    Returns
+    -------
+    Mantid workspace containing the original sample or a default cylinder sample.
+
+    """
+    sample = ws.sample()   
+    shape_xml = sample.getShape().getShapeXML()
+    import xml.etree.ElementTree as etree
+    root = etree.ElementTree(etree.fromstring(shape_xml)).getroot()
+    # if no sample defined, the content of the shapexml element is empty.
+    if root.text == '  ':
+        shape = '''<cylinder id="cyl">
+        <centre-of-bottom-base x="0.0" y="0.0" z="0.0" />
+        <axis x="0.0" y="0.2" z="0.0"/>
+        <radius val="1"/>
+        <height val="10"/>
+        </cylinder>'''
+        simpleapi.CreateSampleShape(ws, shape)
+        simpleapi.SetSampleMaterial(ws, ChemicalFormula="V")
+    return ws
+
+def absorption_correction(input_variable, **kwargs):

```

Comment:
- `input_variable` -> `data` (or something like that, it is a data array, not a variable).
- For forwarding arguments to Mantid algorithms we have "standardized" on using a dict named `mantid_args`, instead of just forwarding any keyword arguments.

Reply:
changed to `data_array`. I don't particularly like `data` as it is way too generic.